toolchain/check/handle_function.cpp

```diff
@@ -39,13 +40,25 @@ auto HandleReturnType(Context& context, Parse::ReturnTypeId node_id) -> bool {
   return true;
 }
 
-static auto DiagnoseModifiers(Context& context,
+static auto DiagnoseModifiers(Context& context, bool is_definition,
                               SemIR::NameScopeId target_scope_id)
     -> KeywordModifierSet {
   const Lex::TokenKind decl_kind = Lex::TokenKind::Fn;
   CheckAccessModifiersOnDecl(context, decl_kind, target_scope_id);
+  if (is_definition) {
+    ForbidExternModifierOnDefinition(context, decl_kind);
+  }
+  if (target_scope_id.is_valid()) {
+    auto target_id = context.name_scopes().Get(target_scope_id).inst_id;
+    if (target_id.is_valid() &&
+        !context.insts().Is<SemIR::Namespace>(target_id)) {
+      ForbidModifiersOnDecl(context, KeywordModifierSet::Extern, decl_kind,
+                            " on member functions");
```

Comment:
I don't really have a better idea for how to phrase this, but I think this includes rejecting `extern` nested functions, correct?

Anyways, not really worth changing now, mostly wanted to make sure I understood it correctly.

Reply:
Do you mean function declarations in functions? Those aren't supported right now, they'd be rejected at function body processing.